### ISSUE TITLE
Add aria-pressed to TextEditor so screenreader displays toggle status

### ISSUE THREAD

**Issue by adamfsage** [Sage/carbon#7038]
### Description

As a user using screenreader, I expect the Text Editor buttons (Bold, Italics, etc) to have the toggle status read out to me so I can tell if they are on or off.

Currently, there's no indication for this.

<img width="1503" alt="Screenshot 2024-10-24 at 16 35 57" src="https://github.com/user-attachments/assets/bf67cdf5-2bef-44f6-acd3-84262cb5ca35">

### Reproduction

https://carbon.sage.com/?path=/docs/text-editor--docs

### Steps to reproduce

Open screenreader in Chrome/Safari and have it read out Bold or Italics button in the Text Editor

### JIRA ticket numbers (Sage only)

SBS-109915

### Suggested solution

I'd either recommend finding a way to add aria-pressed=false to the buttons, failing that, get button state and use an aria-label to achieve an identical effect 

### Carbon version

144.1.1

### Relevant browsers

Chrome, Safari

### Relevant OSs

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
FE-6881
